go.mod

```diff
@@ -12,43 +13,47 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/protobuf v1.5.2
 	github.com/google/cadvisor v0.44.2
+	github.com/google/uuid v1.3.0
 	github.com/kubewharf/katalyst-api v0.4.1-0.20240222122824-be538f641f58
 	github.com/montanaflynn/stats v0.7.1
 	github.com/opencontainers/runc v1.1.6
 	github.com/opencontainers/selinux v1.10.0
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.12.1
-	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.32.1
+	github.com/prometheus/client_golang v1.14.0
+	github.com/prometheus/client_model v0.3.0
+	github.com/prometheus/common v0.37.0
 	github.com/slok/kubewebhook v0.11.0
-	github.com/spf13/cobra v1.4.0
+	github.com/spf13/cobra v1.6.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.8.1
 	github.com/vishvananda/netns v0.0.0-20200728191858-db3c7e526aae
+	github.com/volcengine/volc-sdk-golang v1.0.71
 	go.opentelemetry.io/otel v0.20.0
 	go.opentelemetry.io/otel/exporters/metric/prometheus v0.20.0
 	go.opentelemetry.io/otel/metric v0.20.0
 	go.opentelemetry.io/otel/sdk v0.20.0
 	go.opentelemetry.io/otel/sdk/export/metric v0.20.0
 	go.opentelemetry.io/otel/sdk/metric v0.20.0
-	go.uber.org/atomic v1.7.0
+	go.uber.org/atomic v1.9.0
 	golang.org/x/sys v0.7.0
 	golang.org/x/time v0.0.0-20220210224613-90d013bbcef8
-	gonum.org/v1/gonum v0.6.2
+	gonum.org/v1/gonum v0.8.2
 	google.golang.org/grpc v1.51.0
-	k8s.io/api v0.24.16
-	k8s.io/apimachinery v0.24.16
+	gopkg.in/yaml.v3 v3.0.1
+	k8s.io/api v0.26.1
```

Comment:
why is it necessary to upgrade dependencies here? can we just keep the version (for stability)?

Reply:
Because the referenced package uses this version. But actually replace in go.mod is replacing the version back